timbermill-local/src/main/java/com/datorama/oss/timbermill/cron/TasksMergerJobs.java

```diff
@@ -1,49 +1,40 @@
 package com.datorama.oss.timbermill.cron;
 
-import java.util.Map;
+import java.time.ZonedDateTime;
 
 import org.quartz.Job;
+import org.quartz.JobDataMap;
 import org.quartz.JobExecutionContext;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import com.datorama.oss.timbermill.ElasticsearchClient;
 import com.datorama.oss.timbermill.common.ElasticsearchUtil;
-import com.datorama.oss.timbermill.unit.Task;
-import com.google.common.collect.Maps;
 
-import static com.datorama.oss.timbermill.ElasticsearchClient.ALL_TASK_FIELDS;
-import static com.datorama.oss.timbermill.ElasticsearchClient.PARTIALS_QUERY;
-import static org.elasticsearch.common.Strings.EMPTY_ARRAY;
+import kamon.Kamon;
+import kamon.metric.Metric;
+import kamon.metric.Timer;
+
 
 public class TasksMergerJobs implements Job {
 
 	private static final Logger LOG = LoggerFactory.getLogger(TasksMergerJobs.class);
-	private ElasticsearchClient client;
+
+	private final Metric.Timer partialsJobLatency = Kamon.timer("timbermill2.partial.tasks.job.latency.timer");
 
 	@Override public void execute(JobExecutionContext context) {
-		client = (ElasticsearchClient) context.getJobDetail().getJobDataMap().get(ElasticsearchUtil.CLIENT);
-		String currentIndex = client.getCurrentIndex();
-		String previousIndex = client.getOldIndex();
-		if (indexExists(previousIndex)){
+
```

Comment:
Can you add the currentlyRunning mechanism from the persistence jobs? also please add it to all crons so we don't run 2 a t a time.
Just saw that there could be an bug there if an exception was thrown, can you please fix that so when finishing the cron even with an exception the variable will be set back to false?

Reply:
it can also be handled by using @DisallowConcurrentExecution , on the class definition.
The problem in the TasksMergerJob, is that we have one task per environment, and we do want to allow 2 different envs to run together probably, so perhaps implement it with collection of the env names that prevent running the same env job twice.

regarding the fix in the persistence job, I'll change it to the annotation

Reply:
another option, is to use the envSet , like we do on the OrphansAdoptionJob, so we will schedule only one TasksMergerJob once a minute, on all relevant envs